### PR TITLE
Only preventDefault if key is actually handled

### PR DESCRIPTION
**Pull request by rugk**
#### Description

Fixes https://github.com/mitmproxy/mitmproxy/issues/4132

**This is untested**

Note I do not know why you need to return a lambda function there, but I guess this is some React-thing I don't know.

#### Checklist

 - [ ] I have updated tests where applicable.
 - [ ] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: web/src/js/ducks/ui/keyboard.js [hunk 1]
```diff
@@ -124,9 +119,12 @@ export function onKeyDown(e) {
...
             }
 
-
+            // ignore all other keys
             default:
                 return
         }
+        
+        // prevent key press if it was handled
+        e.preventDefault()
```
Comment (rugk) [inline web/src/js/ducks/ui/keyboard.js:128]: Oh I see the problem here. `e` is not defined in this scope as lambda functions create a new scope IIRC.

Okay, dunno how to solve this. Pass `e` into the function or just remove this lambda function somehow and get the other variables somewhere else?

**Comment by Prinzhorn** [comment]
> Note I do not know why you need to return a lambda function there, but I guess this is some React-thing I don't know.

Same. It's a redux thing and I can't wrap my head around it.

I assume it happens asynchronously. `e` is available in this scope, but you need to `preventDefault` synchronously, so it's too late inside the lambda function.

As a quick fix for your issue we could just add the function keys and call it a day, not sure if @mhils would be excited about that. Something like this:

```js
if (e.ctrlKey || e.metaKey || /^F[0-9]+$/.test(e.key)) {
```

**Comment by rugk** [comment]
Ohh yet another workaround. Whitelisting ctrl and meta is already a workaround… :see_no_evil:

**Comment by mhils** [comment]
Yes, welcome to the world of redux, redux-thunk, and pain. Lot of the latter I guess 😄 

The problem is that we need the application state in the key handler, and redux-thunk requires that callback. One workaround would be to pass the entire redux store to [ProxyApp](https://github.com/mitmproxy/mitmproxy/blob/master/web/src/js/components/ProxyApp.jsx) as a prop, but we need to make sure that this doesn't cause a repaint on every state change. It's honestly a long time since I looked at any of this. If someone wants to take a look at this, please do!


I'm closing this here for now, but we can merge a PR that deals with function keys as a workaround if that helps your particular use case. :)

**Pull request closed without merging**